test/functional/feature_pop_shutdown_sync.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014-2019 The Bitcoin Core developers
+# Copyright (c) 2019-2020 Xenios SEZC
+# https://www.veriblock.org
+# Distributed under the MIT software license, see the accompanying
+# file LICENSE or http://www.opensource.org/licenses/mit-license.php.
+
+"""
+
+"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.pop import mine_until_pop_active
+from test_framework.util import assert_raises_rpc_error, connect_nodes, disconnect_nodes
+import time
+
+class PopShutdownSync(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+        self.extra_args = [["-txindex"], ["-txindex"]]
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+        self.skip_if_no_pypoptools()
+
+    def setup_network(self):
+        self.setup_nodes()
+        for node in self.nodes:
+            node.setmocktime(1617025247)
+        mine_until_pop_active(self.nodes[0])
+
+        # all nodes connected and synced
+        for i in range(self.num_nodes - 1):
+            connect_nodes(self.nodes[i + 1], i)
+        self.sync_all()
+
+    def get_best_block(self, node):
+        hash = node.getbestblockhash()
+        return node.getblock(hash)
+
+    def run_test(self):
+        self.sync_all(self.nodes[0:2])
+        lastblock = self.nodes[0].getblockcount()
+        self.log.info("nodes synced with block height %d", lastblock)
+
+        from pypoptools.pypopminer import MockMiner
+        self.apm = MockMiner()
+
+        disconnect_nodes(self.nodes[0], 1)
+        lastblock = self.nodes[1].getblockcount()
+        self.nodes[1].generate(nblocks=1000)
+        self.log.info("node1 disconnected and generating more blocks")
+        self.nodes[1].waitforblockheight(lastblock + 1000)
+        lastblock = self.nodes[1].getblockcount()
+        self.log.info("node1 reached block height %d", lastblock)
+        connect_nodes(self.nodes[0], 1)
+        self.log.info("node1 reconnected")
+        time.sleep(1) # Sleep for 1 second to let headers sync
```

Comment:
dont use sleep.
if you want nodes to sync, use self.sync_all(...)

Reply:
That's the problem, we don't want the sync to complete. We want to give it a little time to start before stopping it. Do you know of a better way to wait for just the headers to sync but before the block data has a chance to finish?

Reply:
Okay, makes sense